### PR TITLE
Tickets/DM-5022: Modernize Python usage

### PR DESCRIPTION
**Pull request by timj**
Modernize the python code in Qserv to Python 2.7 style. Mainly:

* Print function
* `in` rather than `has_key()`
* Octal numbers use "o"
* List comprehensions rather than map/lambda

### CODE REVIEW

File: admin/bin/qserv-admin.py [hunk 1]
```diff
@@ -636,14 +637,14 @@ def main():
             try:
                 parser.parse(cmd)
             except (_ToolError, css.CssError) as e:
-                print "ERROR: ", e.__str__()
+                print("ERROR: ", e.__str__())
```
Comment (andy-slac) [inline admin/bin/qserv-admin.py]: Not related to modernization, but should not use special methods like `__str__` directly, we'll need to make another round of cleanup for those I guess.

File: admin/bin/qserv-admin.py [hunk 2]
```diff
@@ -636,14 +637,14 @@ def main():
...
                 sys.exit(1)
     else:
         try:
             # wait for commands and process
             parser.receiveCommands()
         except(KeyboardInterrupt, SystemExit, EOFError):
-            print ""
+            print("")
```
Comment (andy-slac) [inline admin/bin/qserv-admin.py]: Is not this the same as `print()`?

File: admin/bin/qserv-admin.py (hunk 1 above)
Comment (timj) [inline admin/bin/qserv-admin.py]: I did not spot that. I'll replace them with `str(e)`.

File: admin/bin/qserv-admin.py (hunk 2 above)
Comment (timj) [inline admin/bin/qserv-admin.py]: Yes. Will fix.

File: admin/python/lsst/qserv/admin/nodeMgmt.py [hunk 3]
```diff
@@ -123,11 +123,11 @@ def selectDict(self, state=None, nodeType=None):
         """
 
         # if state is a string make a list out of it
-        if isinstance(state, types.StringTypes):
+        if isinstance(state, str):
```
Comment (andy-slac) [inline admin/python/lsst/qserv/admin/nodeMgmt.py]: Does not this leave out `unicode` in Python2?

Comment (timj) [inline admin/python/lsst/qserv/admin/nodeMgmt.py]: Yes, I had not meant to leave that change in. Sorry. This needs to be dealt properly for python portability (including the option of jut going to python 3).

File: site_scons/site_tools/unittest.py [hunk 4]
```diff
@@ -91,14 +92,14 @@ def __call__(self, target, source, env):
                 shutil.move(out, out + '.failed')
                 msg = '*** Unit test failed, check log file ' + out + '.failed ***'
                 sep = '*' * len(msg)
-                print sep + '\n' + msg + '\n' + sep
+                print(sep + '\n' + msg + '\n' + sep)
```
Comment (andy-slac) [inline site_scons/site_tools/unittest.py]: Should we modernize further:

    print(sep, msg, sep, sep='\n')

Mixing `sep` as argument and keyword argument is confusing, I'd rename `sep` variable to something else

Comment (timj) [inline site_scons/site_tools/unittest.py]: I'm happy to modernize further.

**Pull request merged**